src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || dataName == "")
+    {
+      if (!is.null(dataImportOptions$importLocation) && dataImportOptions$importLocation != "")
+      {
+        basename <- basename(dataImportOptions$importLocation)
+        if (length(basename) > 0)
+        {
+          fileComponents <- unlist(strsplit(basename, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1])
```

Comment:
The default separator for paste is a space, but we probably don't want that here since (I presume) `dataName` is going to become a variable name. Maybe use `-` or `_`?